test/unit/spec/EnrollCall_spec.js

```diff
@@ -263,6 +265,22 @@ function (Q, Okta, OktaAuth, LoginUtil, Util, Form, Beacon, Expect, $sandbox,
           expect(test.form.errorMessage()).toBe('Invalid Phone Number.');
         });
       });
+      itp('triggers an afterError event and does not go to next step if error response', function () {
+        return setupAndSendInvalidCode().then(function (test) {
+          expectCallButton(test);
+          expect(test.afterRenderHandler).toHaveBeenCalledTimes(1);
+          expect(test.afterRenderHandler.calls.allArgs()[0]).toEqual([
+            {
+              controller: 'enroll-call'
+            },
+            jasmine.objectContaining({
```

Comment:
Because we're returning an `Error` type, we need to assert properties on the object via `objectContaining`. Exact matches unfortunately do not work with error types.

Alternatively, we can import in all known error types and then attempt to match, but we're still going to need to assert our known values the same way we're currently doing so.

```javascript
const errorContext = test.afterErrorHandler.calls.allArgs()[0][1]);
expect(errorContext instanceOf SomeError);
expect(errorContext.name).toBe('SomeError');
```

Reply:
@haishengwu-okta: Great points! I should have communicated this earlier.

1. Most errors we're capturing have XHR values. This is the response value via the API, but isn't always available. As an example, some OAuth error will be captured via the URL instead of an XHR request.
   - Another finding (and recorded [here](https://oktainc.atlassian.net/browse/OKTA-200208)), some of our tests aren't using the same behavior as the actual source code. In many places, we're using [`transformErrorXhr`](https://github.com/okta/okta-signin-widget/blob/4e0c5d2823eb6a283f8e6f216eabf7e6eeba1eca/src/util/Util.js#L23-L55) to extract an `errorCause` from the response, and use it for the `errorSummary`. When attempting to assert the `xhr` in some test files, we're going to need to refactor some of the assertions.
2. Correct. OAuth errors within AuthJS and inherently the Widget do not have `xhr` fields. I've noted this property as *optional* in our [README](https://github.com/okta/okta-signin-widget/pull/594/files#diff-04c6e90faac2675aa89e2176d2eec7d8R1345)
3. Yep. I think having all errors be extensions of the `Error` type is the right call. If a developer wants to know which line/file the error occurs, they can always output and parse the `Error.stack` field.

Reply:
1. I guess we shall assert `xhr` if it does exists. e.g in EnrollU2f_spec, it does have `xhr`.
2. It has more fields than we doc'ed (name, message, statusCode, xhr). I'm hesitate we shall populate extra fields than what we doc'ed
3. I doubt developers need the `Error.stack`. When I say "developers" I meant who use pre-built signin-widget and I cant think a reason that they want to know which line/file the error occurs given the `name/message/statusCode/xhr` shall already give sufficient information. 
(I'm pick at we shall do what we said. hah)

thoughts?